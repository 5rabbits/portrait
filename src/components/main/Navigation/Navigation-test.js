import { React, toJSON, mount } from 'test/helper'
import Navigation from 'main/Navigation'

describe('Navigation', () => {
  it('should render with minimum props', () => {
    const component = mount(<Navigation />)

    expect(toJSON(component)).toMatchSnapshot()
  })
})
