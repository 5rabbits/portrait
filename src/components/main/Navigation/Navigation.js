import React from 'react'
import PropTypes from 'prop-types'
import childrenPropType from 'propTypes/children'
import classNames from 'classnames'

/**
 * Creates a navigation bar.
 */
const Navigation = props =>
  <div
    className={classNames('navbar', 'navbar-fixed-top',
      `navbar-${props.theme}`, 'container-scroll', props.className,
    )}
    >
    <div className="container-max">
      <div className="navbar-header">
        <button type="button" className="navbar-toggle collapsed pull-left" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1" aria-expanded="false">
          <span className="sr-only">Toggle navigation</span>
          <i className="fa fa-bars" />
        </button>
        <button type="button" className="navbar-toggle pull-right" aria-expanded="false">
          <span className="sr-only">Toggle navigation</span>
          <i className="fa fa-search" />
        </button>
        {props.brand &&
          <div className="navbar-brand">
            {props.brand}
          </div>
        }
      </div>

      <div className="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
        <ul
          className={classNames('nav navbar-nav', 'scroll-inline', {
            'navbar-nav--align-left': props.align === 'left',
          })}
          >
          {props.children}
        </ul>
        {props.navigationOptions}
      </div>
    </div>
  </div>

Navigation.propTypes = {
  /**
   * Main links alignment.
   */
  align: PropTypes.oneOf(['left', 'center']).isRequired,

  /**
   * Product name or logo.
   */
  brand: PropTypes.node,

  /**
   * One or more NavigationItem that represents the main links.
   */
  children: childrenPropType,

  /**
   * `className` for the outermost wrapper.
   */
  className: PropTypes.string,

  /**
   * Right navbar sections contents.
   */
  navigationOptions: childrenPropType,

  /**
   * Visual theme.
   */
  theme: PropTypes.oneOf(['default', 'inverse']).isRequired,
}

Navigation.defaultProps = {
  align: 'center',
  brand: null,
  children: null,
  className: null,
  navigationOptions: null,
  theme: 'default',
}

export default Navigation
