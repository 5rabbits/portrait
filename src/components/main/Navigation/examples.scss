.component-example--Navigation {
  [data-reactroot] {
    width: 100%;
  }

  .navbar {
    position: static;
  }
}
