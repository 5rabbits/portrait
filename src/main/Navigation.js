import React from 'react'

const Navigation = () =>
  <div className="navbar navbar-inverse navbar-fixed-top">
    <div className="container-max">
      <div className="navbar-header">
        <button type="button" className="navbar-toggle collapsed pull-left" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1" aria-expanded="false">
          <span className="sr-only">Toggle navigation</span>
          <i className="fa fa-bars" />
        </button>
        <button type="button" className="navbar-toggle pull-right" aria-expanded="false">
          <span className="sr-only">Toggle navigation</span>
          <i className="fa fa-search" />
        </button>
        <a className="navbar-brand" href="#">{this.props.brand}</a>
      </div>

      <div className="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
        <ul className="nav navbar-nav">
          {this.props.children}
        </ul>

        <ul className="nav navbar-nav navbar-right">
          <li className="dropdown">
            <a href="#" className="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Perfil <span className="caret" /></a>
            <ul className="dropdown-menu">
              <li><a href="#">Action</a></li>
              <li><a href="#">Another action</a></li>
              <li><a href="#">Something else here</a></li>
              <li role="separator" className="divider" />
              <li><a href="#">Separated link</a></li>
            </ul>
          </li>
        </ul>
      </div>
    </div>
  </div>

export default Navigation
