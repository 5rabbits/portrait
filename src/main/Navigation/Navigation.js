import React, { PropTypes, Component } from 'react'
import childrenPropType from 'propTypes/children'
import cx from 'classnames'
import responsive from 'helpers/responsive'

class Navigation extends Component {
  static propTypes = {
    align: PropTypes.oneOf(['left', 'center', 'right']).isRequired,
    brand: PropTypes.node,
    children: childrenPropType,
    className: PropTypes.string,
    // navigationOptions: childrenPropType,
    theme: PropTypes.oneOf(['default', 'inverse']).isRequired,
    hideTexts: PropTypes.bool,
    collapsed: PropTypes.bool,
  }

  static defaultProps = {
    align: 'center',
    theme: 'default',
  }

  static childContextTypes = {
    hideNavigationItemText: PropTypes.bool,
  }

  getChildContext() {
    return {
      hideNavigationItemText: this.props.hideTexts,
    }
  }

  componentDidMount() {}

  render() {
    const { align, brand, children, theme, className, collapsed } = this.props

    return (
      <div
        className={cx(
          'navigation',
          `navigation--${theme}`,
          `navigation--align-${align}`,
          className,
          {
            'navigation--collapsed': collapsed,
          },
        )}
        >
        {brand &&
          <div className="navigation__brand-container">
            <div className="navigation__brand">
              {brand}
            </div>
          </div>
        }

        {!collapsed &&
          <div className="navigation__links">
            {children}
          </div>
        }
      </div>
    )
  }
}

const rules = () => ({
  hideTexts: '(max-width: 1000px)',
  collapsed: '(max-width: 800px)',
})

export default responsive(rules)(Navigation)

  // <div
  //   className={classNames('navbar', 'navbar-fixed-top',
  //     `navbar-${props.theme}`, props.className,
  //   )}
  //   >
  //   <div className="container-max">
  //     <div className="navbar-header">
  //       <button type="button" className="navbar-t
  // oggle collapsed pull-left" data-toggle="collapse" d
  // ata-target="#bs-example-navbar-collapse-1" aria-expanded="false">
  //         <span className="sr-only">Toggle navigation</span>
  //         <i className="fa fa-bars" />
  //       </button>
  //       {props.brand &&
  //         <div className="navbar-brand">
  //           {props.brand}
  //         </div>
  //       }
  //     </div>
  //
  //     <div className="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
  //       <ul
  //         className={classNames('nav navbar-nav', {
  //           'navbar-nav--align-left': props.align === 'left',
  //         })}
  //         >
  //         {props.children}
  //       </ul>
  //       {props.navigationOptions}
  //     </div>
  //   </div>
  // </div>
