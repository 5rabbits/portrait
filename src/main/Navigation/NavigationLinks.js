import React, { PropTypes, Component } from 'react'
import NavigationItem from 'main/NavigationItem'
import childrenPropType from 'propTypes/children'
import cx from 'classnames'

export default class NavigationLinks extends Component {
  static propTypes = {
    children: childrenPropType,
    className: PropTypes.string,
    collapsed: PropTypes.bool,
    currentUser: PropTypes.shape({
      name: PropTypes.string.isRequired,
      image: PropTypes.string,
    }),
    userLinks: childrenPropType,
    hideTexts: PropTypes.bool,
  }

  static childContextTypes = {
    hideNavigationItemText: PropTypes.bool,
  }

  getChildContext() {
    return {
      hideNavigationItemText: this.props.hideTexts,
    }
  }

  render() {
    const { children, currentUser, collapsed, userLinks, className } = this.props

    return (
      <div className={cx('navigation__links', className)}>
        <div className="navigation__main-links">
          {children}
        </div>

        <div className="navigation__user-links">
          {currentUser &&
            <NavigationItem
              hideText={!collapsed}
              label={currentUser.name}
              icon={
                <img
                  alt={currentUser.name}
                  className="navigation__user-image"
                  src={currentUser.image}
                />
              }
              menuAlign="right"
              >
              {userLinks}
            </NavigationItem>
          }
        </div>
      </div>
    )
  }
}
