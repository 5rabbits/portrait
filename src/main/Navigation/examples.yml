-
  description: Default theme.
  code: >
    <Navigation brand="Example">
      <NavigationItem label="Link1" />
      <NavigationItem label="Link2" />
      <NavigationItem label="Link3" />
    </Navigation>

-
  description: Inverse theme.
  code: >
    <Navigation brand="Example" theme="inverse">
      <NavigationItem label="Link1" />
      <NavigationItem label="Link2" />
      <NavigationItem label="Link3" />
    </Navigation>
