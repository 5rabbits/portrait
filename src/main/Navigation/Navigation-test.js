import { React, expect, mount } from 'test/helper'
import Navigation from 'main/Navigation'

describe('Navigation', () => {
  it('should render with minimum props', () => {
    const component = mount(<Navigation />)

    expect(component).to.be.ok
  })
})
