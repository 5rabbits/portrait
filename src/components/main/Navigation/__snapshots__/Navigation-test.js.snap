// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should render the brand if specified 1`] = `
<Navigation
  align="center"
  brand="TEST"
  className={null}
  navigationOptions={null}
  theme="default"
>
  <div
    className="navbar navbar-fixed-top navbar-default container-scroll"
  >
    <div
      className="container-max"
    >
      <div
        className="navbar-header"
      >
        <button
          aria-expanded="false"
          className="navbar-toggle collapsed pull-left align-left"
          data-target="#bs-example-navbar-collapse-1"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="sr-only"
          >
            Toggle navigation
          </span>
          <i
            className="fa fa-bars"
          />
        </button>
        <button
          aria-expanded="false"
          className="navbar-toggle pull-right align-right"
          type="button"
        >
          <span
            className="sr-only"
          >
            Toggle navigation
          </span>
          <i
            className="fa fa-search"
          />
        </button>
        <div
          className="navbar-brand"
        >
          TEST
        </div>
      </div>
      <div
        className="collapse navbar-collapse"
        id="bs-example-navbar-collapse-1"
      >
        <ul
          className="nav navbar-nav scroll-inline"
        />
      </div>
    </div>
  </div>
</Navigation>
`;

exports[`Navigation should render with minimum props 1`] = `
<Navigation
  align="center"
  brand={null}
  className={null}
  navigationOptions={null}
  theme="default"
>
  <div
    className="navbar navbar-fixed-top navbar-default container-scroll"
  >
    <div
      className="container-max"
    >
      <div
        className="navbar-header"
      >
        <button
          aria-expanded="false"
          className="navbar-toggle collapsed pull-left align-left"
          data-target="#bs-example-navbar-collapse-1"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="sr-only"
          >
            Toggle navigation
          </span>
          <i
            className="fa fa-bars"
          />
        </button>
        <button
          aria-expanded="false"
          className="navbar-toggle pull-right align-right"
          type="button"
        >
          <span
            className="sr-only"
          >
            Toggle navigation
          </span>
          <i
            className="fa fa-search"
          />
        </button>
      </div>
      <div
        className="collapse navbar-collapse"
        id="bs-example-navbar-collapse-1"
      >
        <ul
          className="nav navbar-nav scroll-inline"
        />
      </div>
    </div>
  </div>
</Navigation>
`;
